src/SystemClock.cs

```diff
@@ -5,30 +5,16 @@ namespace Archon
 	/// <summary>
 	/// Aids in unit testing by providing a mockable way to get the current time.
 	/// </summary>
-	public static class SystemClock
+	public interface SystemClock
```

Comment:
Interfaces should have names prefixed with `I`

Reply:
I subscribe to the school of thought that despises Hungarian notation.

Reply:
What of [Microsoft's naming guidelines](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/names-of-classes-structs-and-interfaces)?

Reply:
FYI, I find Hungarian notation to be:

- Grossly misunderstood (it was intended to indicate kind, not type)
- Rarely useful in the first place

However, I find consistency to be:

- Better than any individual style by a large margin

Reply:
I agree with you in principle, but I still despise the `I` prefix.

One argument that may or may not convince you:

> Not having the “I” prefix helped force me to come up with better names for my implementations. For example, instead of just having `IThingDoer` (interface) and `ThingDoer` (implementation), I would instead try to go with something like `ThingDoer` (interface) and `ThingDoerThatDoesItACertainWay` (implementation).

[From here](https://jakejgordon.wordpress.com/2014/10/13/idiomatic-c-sharp-and-the-i-prefix/).

Reply:
And I agree that `ThingDoerThatDoesItACertainWay` is a convention worth following for any implementation. We can, and IMO ought to, enforce this through code reviews.

I just happen to also think that consistency with the rest of the .NET ecosystem has an intrinsic value by merit of reducing mental overhead. Since you can't rename `IEnumerable` or the other thousands of interfaces in .NET, the only decision left to make is whether to be consistent or not.